SourceX/miniwin/ddraw.cpp

```diff
@@ -88,10 +100,17 @@ HRESULT StubDraw::CreateSurface(LPDDSURFACEDESC lpDDSurfaceDesc, LPDIRECTDRAWSUR
 
 	gpBuffer = (BYTE *)pal_surface->pixels; // Hack should happen in dx.cpp, but gives error
 
+#ifdef USE_SDL1
+	if (SDL_SetPalette(pal_surface, SDL_LOGPAL, palette->colors, 0, palette->ncolors) != 1) {
```

Comment:
What prevents you from making a SDL_SetSurfacePalette like wrapper for this?

Reply:
It'd be misleading, as the 2 functions are quite different:

SDL1: Palette data is copied to the Surface and the caller needs to set SDL_LOG/PHYSPAL flags correctly depending on the surface type.

SDL2: Palette *pointer* is copied to the Surface, but the caller still owns the Palette pointer and can modify the surface's palette through it directly.

Reply:
This is also the reason for the `#ifdef USE_SDL1` in `SDrawUpdatePalette`.